### ISSUE TITLE
🐛 Crash in android when app background. 

### ISSUE THREAD

**Issue by e-vural** [cuvent/react-native-vision-camera#2516]
### What's happening?

- Start video recording
- App move to background
- App crashed

I try to run your example but camera is frozen when app open. 

### Reproduceable Code

```tsx
const isFocused = useIsFocused()
    const appState = useAppState()
    const isActive = isFocused && appState === "active"

    useEffect(() => {
        setTimeout(() => {
            setCameraIsActive(true);
        }, 10)

    }, []);

{
                cameraIsActive &&
            
                <View style={{
                    position:"absolute",
                    aspectRatio: 9 / 16,
                    width: wp(100) - (isFullyFit ? 0.4 : 0) }}//Samsung Note FE dialog problem
                >

                    <Camera
                        onError={(er) => {
                            console.log("camera error ",er)}}
                        onStopped={(e) => {
                            console.log("onStopped",e)
                        }}
                        exposure={0}
                        isActive={isActive}
                        device={device}
                        ref={camera}
                        format={format}
                        video={true}
                        audio={true}
                        enableBufferCompression={true}
                        // resizeMode={"contain"}
                        style={{
                            aspectRatio: 9 / 16,
                            zIndex: -1,
                            width: "100%",
                            height: "100%",
                            alignSelf: "center",
                        }}
                    />

                </View>

}
```

### Relevant log output

```shell
FATAL EXCEPTION: main
                                                                                                    Process: com.onremoteassistant.android.dev, PID: 475
                                                                                                    java.lang.RuntimeException: Cannot render to a null Surface! Error: 12291
                                                                                                    	at com.mrousavy.camera.core.VideoPipeline.onFrame(Native Method)
                                                                                                    	at com.mrousavy.camera.core.VideoPipeline.onFrameAvailable(VideoPipeline.kt:170)
                                                                                                    	at android.graphics.SurfaceTexture$1.handleMessage(SurfaceTexture.java:206)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loop(Looper.java:214)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7050)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
```

### Camera Device

```json
{
  "formats": [],
  "sensorOrientation": "landscape-right",
  "hardwareLevel": "full",
  "maxZoom": 8,
  "minZoom": 1,
  "maxExposure": 20,
  "supportsLowLightBoost": false,
  "neutralZoom": 1,
  "physicalDevices": [
    "wide-angle-camera"
  ],
  "supportsFocus": true,
  "supportsRawCapture": true,
  "isMultiCam": false,
  "minFocusDistance": 10,
  "minExposure": -20,
  "name": "BACK (0)",
  "hasFlash": true,
  "hasTorch": true,
  "position": "back",
  "id": "0"
}
```

### Device

Samsung Note FE, Samsung S21 , Xiaomi Redmi Pro 11

### VisionCamera Version

3.8.2

### Can you reproduce this issue in the VisionCamera Example app?

I didn't try (⚠️ your issue might get ignored & closed if you don't try this)

### Additional information

- [ ] I am using Expo
- [ ] I have enabled Frame Processors (react-native-worklets-core)
- [x] I have read the [Troubleshooting Guide](https://react-native-vision-camera.com/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

**Comment by mrousavy** [comment]
Hey thank you for your bug report!

I just released [VisionCamera 3.9.0-beta.1](https://github.com/mrousavy/react-native-vision-camera/releases/tag/v3.9.0-beta.1), which includes the [big `PersistentCaptureSession` PR](https://github.com/mrousavy/react-native-vision-camera/pull/2494) that aims to fix all the blackscreen issues.... This also includes a fix for the issue you were seeing! :eyes::tada: 

You can try that new beta out yourself and let me know if you still encounter issues. I'll continue to test on more devices in the meantime, organize the code a bit better, and then I'll tackle the preview stretching, focus, and flash issues!

If that fixes the issue for you, please consider [sponsoring me on GitHub](https://github.com/sponsors/mrousavy) to say thanks! ❤️

**Comment by e-vural** [comment]
@mrousavy I tested on [3.9.0-beta.1](https://github.com/mrousavy/react-native-vision-camera/compare/v3.9.0-beta.0...v3.9.0-beta.1) and works fine on android and IOS . IOS has diffrent problem about pause and resume when app goes to background. I guess that issue already created
